packages/quantic/force-app/main/default/lwc/quanticFacet/e2e/quanticFacet.e2e.ts

```diff
@@ -0,0 +1,209 @@
+import {testSearch, testInsight, expect} from './fixture';
+import {
+  useCaseEnum,
+  useCaseTestCases,
+} from '../../../../../../playwright/utils/useCase';
+import facetData from './data';
+
+const fixtures = {
+  search: testSearch as typeof testSearch,
+  insight: testInsight as typeof testInsight,
+};
+
+useCaseTestCases.forEach((useCase) => {
+  let test;
+  if (useCase.value === useCaseEnum.search) {
+    test = fixtures[useCase.value] as typeof testSearch;
+  } else {
+    test = fixtures[useCase.value] as typeof testInsight;
+  }
+
+  test.describe(`quantic facet ${useCase.label}`, () => {
+    test.describe('when selecting and deselecting a facet value', () => {
+      test('should trigger a new search and log the corresponding UA analytics events', async ({
+        search,
+        facet,
+      }) => {
+        const selectedIndex = 0;
+        const {facetId, field, values} = facetData;
+        const selectUaRequest = facet.waitForFacetSelectUaAnalytics({
+          facetId,
+          facetField: field,
+          facetValue: values[selectedIndex].value,
+        });
+        let searchResponsePromise = search.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+        await selectUaRequest;
+        let searchResponse = await searchResponsePromise;
+
+        const {analytics: analyticsForFacetSelect} =
+          facet.extractDataFromSearchResponse(searchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+
+        const deselectUaRequest = facet.waitForFacetDeselectUaAnalytics({
+          facetId,
+          facetField: field,
+          facetValue: values[selectedIndex].value,
+        });
+        searchResponsePromise = search.waitForSearchResponse();
+        await facet.clickOnFacetValue(selectedIndex);
+        await deselectUaRequest;
+        searchResponse = await searchResponsePromise;
```

Comment:
```suggestion
        const deselectUaRequest = facet.waitForFacetDeselectUaAnalytics({
          facetId,
          facetField: field,
          facetValue: values[selectedIndex].value,
        });
        searchResponsePromise = search.waitForSearchResponse();

        await facet.clickOnFacetValue(selectedIndex);
        await deselectUaRequest;
        searchResponse = await searchResponsePromise;
```

for better readability/consistency